modules/objdetect/src/qrcode.cpp

```diff
@@ -2251,6 +2251,90 @@ bool QRDecode::preparingCurvedQRCodes()
     return true;
 }
 
+/**
+ * @param finderPattern 4 points of finder pattern markers, calculated by findPatternsVerticesPoints()
+ * @return true if the pattern has the correct side lengths
+ */
+inline bool checkFinderPatternByAspect(const vector<Point> &finderPattern) {
+    if (finderPattern.size() != 4ull)
+        return false;
+    vector<float> sidesLen;
+    for (size_t i = 0; i < finderPattern.size(); i++) {
+        sidesLen.push_back(sqrt(normL2Sqr<float>(Point2f(finderPattern[i] - finderPattern[(i+1ull)%finderPattern.size()]))));
+    }
+    std::sort(sidesLen.begin(), sidesLen.end());
```

Comment:
in case we know, that we have 4 length values min and max are obvious without sort: `max(max(a1, a2), max(a3, a4))`.

Reply:
fixed